File: 1a.php
<?php

$fin = fopen("1.dat", "r");
$data = fread($fin, filesize("1.dat"));
$floor = 0;
for ($i = 0; $i < strlen($data); $i++) {
    if ($data[$i] === "(") {
        $floor++;
    }
    if ($data[$i] === ")") {
        $floor--;
    }
}
echo $floor;
fclose($fin);

?>

File: 1a.cc
#include <fstream>
#include <iostream>

int main(void) {
    std::string line;
    std::ifstream fin;
    int floor = 0;

    fin.open("1.dat");
    std::getline(fin, line);
    for (auto c : line) {
        if (c == '(')
            floor++;
        if (c == ')')
            floor--;
    }
    std::cout << floor << std::endl;
}
